### ISSUE TITLE
Installation issue: gtkplus

### ISSUE THREAD

**Issue by cessenat** [LLNL/spack#23348]
```console
$ spack install octave+qt+qscintilla+fftw+fontconfig+freetype+glpk+gl2ps+magick+gnuplot+zlib+curl+fltk+qrupdate %gcc ^mpich ^openblas
...
ninja: build stopped: subcommand failed.
  >> 429    FAILED: gdk/Gdk-3.0.gir
```

### Information on your system

* **Spack:** 0.16.1-2442-6b12000
* **Python:** 3.6.8
* **Platform:** linux-centos7-skylake
* **Concretizer:** original

```
$ uname -a
Linux vpn-linux 3.10.0-1127.19.1.el7.x86_64 #1 SMP Tue Aug 25 17:23:54 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```

```
$ cat /etc/redhat-release 
CentOS Linux release 7.8.2003 (Core)
```

No ~/.spack/packages.yaml

Using gcc@8.4.0

### Additional information

* [spack-build-out.txt](https://github.com/spack/spack/files/6401209/spack-build-out.txt)
* [spack-build-out.txt](https://github.com/spack/spack/files/6401210/spack-build-out.txt)

### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and @mentioned any maintainers
- [x] I have uploaded the build log and environment files
- [ ] I have searched the issues of this repo and believe this is not a duplicate

**Comment by mpokorny** [comment]
I can reproduce this issue in gtkplus@3.24.29 (the same error message, that is) on RHEL7, gcc 11, with the Spack develop branch. I can also confirm that this issue does not appear to be a duplicate of any other issue in the Spack repo (_i.e_ check that fourth box).

**Comment by mpokorny** [comment]
I believe this issue may have been solved by #25488. With that PR, my build succeeds. Also having reviewed the build output attached to this issue, it certainly seems likely that #25488 would have fixed it.

**Comment by cessenat** [comment]
Thanks @mpokorny , it is solved.
I encoutered a failure building qscintilla, so I added a more explicit recipe for it:
`spack install octave+qt+qscintilla+fftw+fontconfig+freetype+glpk+gl2ps+magick+gnuplot+zlib+curl+fltk+qrupdate %gcc ^mpich ^openblas ^qscintilla+python+designer ^qt+dbus~framework`
It provided octave with following guis:
```
$ octave --gui
>> available_graphics_toolkits()
ans =
{
  [1,1] = fltk
  [1,2] = gnuplot
  [1,3] = qt
}

```

Fine now.
